api/login_api.py

```diff
@@ -34,12 +32,14 @@ def do_post(self):
     message = "Unable to Authenticate"
 
     try:
-      unused_idinfo = id_token.verify_oauth2_token(
+      idinfo = id_token.verify_oauth2_token(
           token, requests.Request(),
           settings.GOOGLE_SIGN_IN_CLIENT_ID)
-      # userid = idinfo['sub']
-      # email = idinfo['email']
-      session["id_token"] = token
+      user_info = {
+          'email': idinfo['email'],
+          }
+      signature = xsrf.generate_token(str(user_info))
+      session['signed_user_info'] = user_info, signature
```

Comment:
How often does this signature expire?

Reply:
10 days.  It is determined by the timeout specified when we verify the signature in `framework/users.py`.

(And, an implication of having it limited rather than infinite is that we should some kind of refresh logic, otherwise a user could be using our site for 10 days and then be suddenly signed out.   I think that we can add that into the existing code in `api/token_refresh_api.py` rather than trying to get a new id_token like we did with "sign-in with google".  That can be part of a future PR.)

Reply:
Whoops.  I was trying to address this comment by adding a 10 day expiration to the session cookie and I push that change to main rather than to this branch.  Please consider it to be part of this PR.  The commit is here:
https://github.com/GoogleChrome/chromium-dashboard/commit/cc182d544e75f0fd9e9e0494790c1f7cfbffae0a